libyul/backends/evm/AbstractAssembly.h

```diff
@@ -56,7 +57,7 @@ class AbstractAssembly
 	virtual ~AbstractAssembly() = default;
 
 	/// Set a new source location valid starting from the next instruction.
-	virtual void setSourceLocation(langutil::SourceLocation const& _location) = 0;
+	virtual void setSourceLocation(std::shared_ptr<DebugData const> const& _debugData) = 0;
```

Comment:
I do not think it is a good idea keeping this named `setSourceLocation` yet operating on DebugData. I would say we should not change this in this PR, but rather postpone it to the one where we introduce `originLocation`. Then we'll need to think about what is the correct logic and implement it.

Reply:
Makes sense, will leave as it was, and later we can see how we can change it